### PR TITLE
Move 'generate XXX' code fixers down to code-style layer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@davidbritch @ToddGrun ptal.

File: src/Analyzers/CSharp/Tests/GenerateConstructor/GenerateConstructorTests.cs [hunk 1]
```diff
@@ -2788,74 +2784,6 @@ class D
             """);
     }
 
-    [Fact, WorkItem("http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/530003")]
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/GenerateConstructor/GenerateConstructorTests.cs:2791]: these teests moved to the end.  they're enabled for non-CODESTYLE layer as they depend on humanizer (which codestyle doesn't ship with).

File: src/Analyzers/Core/CodeFixes/CodeFixesResources.resx [hunk 2]
```diff
@@ -267,4 +267,55 @@
   <data name="TODO_colon_override_finalizer_only_if_0_has_code_to_free_unmanaged_resources" xml:space="preserve">
     <value>TODO: override finalizer only if '{0}' has code to free unmanaged resources</value>
   </data>
+  <data name="Generate_constant_0" xml:space="preserve">
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/CodeFixesResources.resx:270]: lots of resource moves.  it's hte majority of hte diff.

File: src/Analyzers/Core/CodeFixes/GenerateConstructor/AbstractGenerateConstructorService.State.cs [hunk 3]
```diff
@@ -574,7 +585,12 @@ private IEnumerable<string> GetUnavailableMemberNames()
 
             Contract.ThrowIfNull(TypeToGenerateIn);
 
+#if CODE_STYLE
+            var provider = document.Project.Solution.Workspace.Services.GetExtendedLanguageServices(TypeToGenerateIn.Language);
+#else
             var provider = document.Project.Solution.Services.GetLanguageServices(TypeToGenerateIn.Language);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/GenerateConstructor/AbstractGenerateConstructorService.State.cs:591]: @tmat would love a way to abstract this behind something (but these helpers have genuinely different types).   i'll look into this myself.  But maybe you have some good ideas here.  This pattern shows up in a lot of files.

**Review by dibarbet — APPROVED** [review]

**Pull request merged**